// react-img-carousel/lib/stories/index.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.rightAlignedSlides = exports.leftAlignedSlides = exports.addImages = exports.customDotsComponent = exports.backgroundImagesWithFade = exports.autoplayWithBackgroundImages = exports.infiniteWithOnly1Slide = exports.infiniteWithOnly2Slides = exports.fadeTransition = exports.nonInfiniteWithCellPadding = exports.infiniteWithCellPadding = exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _index = _interopRequireDefault(require("../index"));

require('../carousel.less');

var _default = {
  component: _index.default,
  title: 'Carousel'
};
exports.default = _default;
const IMAGES = ['http://picsum.photos/400/300', 'http://picsum.photos/275/300', 'http://picsum.photos/400/300', 'http://picsum.photos/350/300', 'http://picsum.photos/250/300', 'http://picsum.photos/375/300', 'http://picsum.photos/425/300', 'http://picsum.photos/325/300'];
const imgElements = IMAGES.map((image, index) => /*#__PURE__*/_react.default.createElement("img", {
  src: image,
  key: index,
  alt: "A sample"
}));

const CustomDots = ({
  numSlides,
  selectedIndex,
  goToSlide,
  title
}) => {
  const dots = [];

  for (let index = 0; index < numSlides; index++) {
    const buttonStyle = {
      border: 'none',
      cursor: 'pointer',
      background: 'transparent'
    };

    if (index === selectedIndex) {
      buttonStyle.color = 'red';
    }

    dots.push( /*#__PURE__*/_react.default.createElement("li", {
      key: `dot-${index}`,
      style: {
        display: 'inline-block'
      }
    }, /*#__PURE__*/_react.default.createElement("button", {
      style: buttonStyle,
      onClick: goToSlide.bind(null, index)
    }, "\u2022")));
  }

  return /*#__PURE__*/_react.default.createElement("div", {
    style: {
      position: 'absolute',
      top: '10px',
      right: '10px',
      background: 'rgba(114, 114, 114, 0.6)',
      zIndex: '1'
    }
  }, /*#__PURE__*/_react.default.createElement("h2", null, title), /*#__PURE__*/_react.default.createElement("ul", {
    style: {
      listStyle: 'none',
      padding: '0'
    }
  }, dots));
};

CustomDots.propTypes = {
  numSlides: _propTypes.default.number.isRequired,
  selectedIndex: _propTypes.default.number,
  goToSlide: _propTypes.default.func.isRequired,
  title: _propTypes.default.node
};

const infiniteWithCellPadding = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  cellPadding: 5
}, imgElements);

exports.infiniteWithCellPadding = infiniteWithCellPadding;

const nonInfiniteWithCellPadding = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  infinite: false,
  width: "450px",
  cellPadding: 5
}, imgElements);

exports.nonInfiniteWithCellPadding = nonInfiniteWithCellPadding;

const fadeTransition = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  slideHeight: "300px",
  transitionDuration: 1000,
  draggable: false,
  transition: "fade"
}, imgElements);

exports.fadeTransition = fadeTransition;

const infiniteWithOnly2Slides = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  arrows: false,
  slideHeight: "300px"
}, /*#__PURE__*/_react.default.createElement("img", {
  src: "http://picsum.photos/325/300",
  alt: "A sample"
}), /*#__PURE__*/_react.default.createElement("img", {
  src: "http://picsum.photos/350/300",
  alt: "A sample"
}));

exports.infiniteWithOnly2Slides = infiniteWithOnly2Slides;

const infiniteWithOnly1Slide = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  infinite: true,
  arrows: false,
  dots: false
}, /*#__PURE__*/_react.default.createElement("img", {
  src: "http://picsum.photos/325/300",
  alt: "A sample"
}));

exports.infiniteWithOnly1Slide = infiniteWithOnly1Slide;

const autoplayWithBackgroundImages = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "100%",
  slideWidth: "100%",
  slideHeight: "70vh",
  arrows: false,
  autoplay: true
}, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/600/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/650/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/675/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/700/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}));

exports.autoplayWithBackgroundImages = autoplayWithBackgroundImages;

const backgroundImagesWithFade = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "100%",
  slideWidth: "100%",
  slideHeight: "70vh",
  transition: "fade",
  transitionDuration: 1000,
  autoplay: true,
  arrows: true
}, /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/600/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/650/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/675/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/700/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/750/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/725/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}), /*#__PURE__*/_react.default.createElement("div", {
  style: {
    backgroundImage: 'url(http://picsum.photos/625/300)',
    backgroundSize: 'cover',
    height: '100%',
    width: '100%'
  }
}));

exports.backgroundImagesWithFade = backgroundImagesWithFade;

const customDotsComponent = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  cellPadding: 5,
  infinite: false,
  dots: false,
  arrows: false,
  autoplay: true,
  controls: [{
    component: CustomDots,
    props: {
      title: 'My Slides'
    },
    position: 'top'
  }],
  transitionDuration: 0
}, imgElements);

exports.customDotsComponent = customDotsComponent;

const addImages = () => {
  const [images, setImages] = (0, _react.useState)([IMAGES[0]]);

  const addImage = () => {
    if (images.length < IMAGES.length) {
      setImages(images.concat(IMAGES[images.length]));
    }
  };

  return /*#__PURE__*/_react.default.createElement(_react.Fragment, null, /*#__PURE__*/_react.default.createElement(_index.default, {
    width: "450px",
    cellPadding: 5,
    infinite: true,
    dots: false,
    arrows: false,
    autoplay: false,
    controls: [{
      component: CustomDots,
      props: {
        title: 'My Slides'
      },
      position: 'top'
    }]
  }, images.map((image, index) => /*#__PURE__*/_react.default.createElement("img", {
    key: index,
    src: image,
    alt: "A sample"
  }))), /*#__PURE__*/_react.default.createElement("button", {
    onClick: addImage
  }, "Add Image"));
};

exports.addImages = addImages;

const leftAlignedSlides = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  cellPadding: 5,
  slideAlignment: "left"
}, imgElements);

exports.leftAlignedSlides = leftAlignedSlides;

const rightAlignedSlides = () => /*#__PURE__*/_react.default.createElement(_index.default, {
  width: "450px",
  cellPadding: 5,
  slideAlignment: "right"
}, imgElements);

exports.rightAlignedSlides = rightAlignedSlides;
